### PR TITLE
Removed dependency on native tar command and used libarchive library

### PR DESCRIPTION
**Pull request by albuquerque**
Replaced call to tar utility and added support for libarchive library instead as we were having issues with tar hanging up infinitely for EOF forcing us to restart iceccd. This gives the ability to use all environments supported by libarchive without having to depend on an external utility.

Also added more descriptive log messages.

### CODE REVIEW

File: daemon/environment.cpp [hunk 1]
```diff
@@ -109,7 +112,7 @@ static bool exec_and_wait(const char *const argv[])
         // parent
         int status;
 
-        while (waitpid(pid, &status, 0) < 0 && errno == EINTR) {}
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: I like the old way better.  They are equivalent, but the {} makes it clear that the loop doing nothing is intentional.

File: daemon/environment.cpp [hunk 2]
```diff
@@ -398,7 +401,7 @@ size_t finish_create_env(int pipe, const string &basedir, string &native_environ
     char buf[1024];
     buf[0] = '\0';
 
-    while (read(pipe, buf, 1023) < 0 && errno == EINTR) {}
+    while (read(pipe, buf, 1023) < 0 && errno == EINTR);
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: As above, {} is clearer in intent.

File: daemon/environment.cpp [hunk 3]
```diff
@@ -425,11 +428,37 @@ size_t finish_create_env(int pipe, const string &basedir, string &native_environ
...
+        }
+    }
+}
+
 pid_t start_install_environment(const std::string &basename, const std::string &target,
                                 const std::string &name, MsgChannel *c,
                                 int &pipe_to_stdin, FileChunkMsg *&fmsg,
                                 uid_t user_uid, gid_t user_gid, int extract_priority)
 {
+    log_error() << "start_install_environment: " << basename << " target "<<target << " Name: " << name << endl;
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: log_error is the wrong level for this. You want log_info

File: daemon/environment.cpp [hunk 4]
```diff
@@ -540,32 +557,81 @@ pid_t start_install_environment(const std::string &basename, const std::string &
...
+
+    a=archive_read_new();
+    archive_read_support_format_all(a);
+    archive_read_support_filter_all(a);
+    ext = archive_write_disk_new();
+    archive_write_disk_set_options(ext, flags);
+    archive_write_disk_set_standard_lookup(ext);
+
+    if(archive_read_open_fd(a, 0, fmsg->len)){
+        trace() << "start_install_environment: "<< "archive_read_open_fd faied"<< endl;
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: This should be log_error

File: daemon/environment.cpp [hunk 5]
```diff
@@ -540,32 +557,81 @@ pid_t start_install_environment(const std::string &basename, const std::string &
...
+    }
+
+    for(;;){
+        int r = archive_read_next_header(a, &entry);
+        if (r == ARCHIVE_EOF) {
+            trace() << "start_install_environment: reached end of archive while recveiving environment "<< endl;
+            break;
+        }
+        if (r < ARCHIVE_WARN){
+            trace() << "start_install_environment: r  < ARCHIVE_WARN " <<archive_error_string(a)<<endl;
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: log_error

File: daemon/environment.cpp [hunk 6]
```diff
@@ -540,32 +557,81 @@ pid_t start_install_environment(const std::string &basename, const std::string &
...
+        /*Extracting archive*/
+        const char* currentFile = archive_entry_pathname(entry);
+        const std::string fullOutputPath = dirname + "/"+currentFile;
+        archive_entry_set_pathname(entry, fullOutputPath.c_str());
+        r = archive_write_header(ext, entry);
+
+        if(archive_entry_size(entry) > 0){
+            r= copy_data(a, ext);
+            if(r < ARCHIVE_WARN){
+                trace()<< "start_install_environment: " << archive_error_string(ext)<<endl;
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: isn't this an error - use log_error

File: daemon/environment.cpp [hunk 7]
```diff
@@ -540,32 +557,81 @@ pid_t start_install_environment(const std::string &basename, const std::string &
...
+                _exit(1);
+            }
+        }
+        r=archive_write_finish_entry(ext);
+        if(r<ARCHIVE_WARN){
+            _exit(1);
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: shouldn't we log an error here?

File: daemon/environment.cpp [hunk 8]
```diff
@@ -614,7 +678,7 @@ size_t remove_environment(const string &basename, const string &env)
     if (pid) {
         int status = 0;
 
-        while (waitpid(pid, &status, 0) < 0 && errno == EINTR) {}
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: I prefer {}

File: daemon/environment.cpp (hunk 1 above)
Comment (albuquerque) [reply daemon/environment.cpp]: done.

File: daemon/environment.cpp (hunk 2 above)
Comment (albuquerque) [reply daemon/environment.cpp]: done.

File: daemon/environment.cpp (hunk 4 above)
Comment (albuquerque) [reply daemon/environment.cpp]: done

File: daemon/environment.cpp (hunk 6 above)
Comment (albuquerque) [reply daemon/environment.cpp]: done

File: daemon/environment.cpp (hunk 8 above)
Comment (albuquerque) [reply daemon/environment.cpp]: done

**Comment by albuquerque** [comment]
As requested, I have updated the NEWS and README files as well in the latest commit.

**Comment by HenryMiller1** [comment]
Next release will probably be 1.3, but that can be considered latter.

**Comment by HenryMiller1** [comment]
Thanks for your work.

**Pull request merged**